Mage/src/main/java/mage/abilities/keyword/ReconfigureAbility.java

```diff
@@ -17,22 +18,33 @@
  */
 public class ReconfigureAbility extends ActivatedAbilityImpl {
 
-    private final String manaString;
+    private final Cost cost;
+//    private final String manaString;
 
     public ReconfigureAbility(String manaString) {
         super(Zone.BATTLEFIELD, new AttachEffect(Outcome.BoostCreature), new ManaCostsImpl<>(manaString));
-        this.manaString = manaString;
+        this.cost = new ManaCostsImpl<>(manaString);
+        this.addTarget(new TargetControlledCreaturePermanent());
+        this.addSubAbility(new ReconfigureUnattachAbility(cost));
+        Ability ability = new SimpleStaticAbility(new ReconfigureTypeEffect());
+        ability.setRuleVisible(false);
+        this.addSubAbility(ability);
+    }
```

Comment:
Don't duplicate the constructor logic. Instead, call `this(new ManaCostsImpl<>(manaString))`

Reply:
Much nicer way of doing it, I'm still getting used to Java